server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -1596,6 +1596,19 @@ private Volume orchestrateAttachVolumeToVM(Long vmId, Long volumeId, Long device
         StoragePoolVO destPrimaryStorage = null;
         if (exstingVolumeOfVm != null && !exstingVolumeOfVm.getState().equals(Volume.State.Allocated)) {
             destPrimaryStorage = _storagePoolDao.findById(exstingVolumeOfVm.getPoolId());
+            if(destPrimaryStorage.getPodId() == null) {
+                destPrimaryStorage.setPodId(vm.getPodIdToDeployIn());
+
+            }
+            if(destPrimaryStorage.getClusterId() == null) {
+                HostVO hostVO = null;
+                if (vm.getHostId() != null) {
+                    hostVO = _hostDao.findById(vm.getHostId());
+                } else {
+                    hostVO = _hostDao.findById(vm.getLastHostId());
+                }
+                destPrimaryStorage.setClusterId(hostVO.getClusterId());
```

Comment:
Do we want to set the clusterID for zonewide storage too?

Reply:
This was added because we were encountering a null pointer exception in some scenarios I don't recall exactly. I added the bit to get the last host id of the vm as attaching a disk to a stopped vm would fail. Have not encountered any side effects with regards to zone wide storage so far..